model/task/task.go

```diff
@@ -2047,21 +2058,47 @@ func (t *Task) makeArchivedTask() *Task {
 
 // Aggregation
 
-// MergeNewTestResults returns the task with both old (embedded in
-// the tasks collection) and new (from the testresults collection) test results
-// merged in the Task's LocalTestResults field.
-func (t *Task) MergeNewTestResults() error {
+// PopulateTestResults returns the task with both old (embedded in the tasks
+// collection) and new (from the testresults collection) OR Cedar test results
+// merged into the Task's LocalTestResults field.
+func (t *Task) PopulateTestResults() error {
+	if t.testResultsPopulated {
```

Comment:
Does it not work to use len(LocalTestResults) > 0 or do you want to avoid repeatedly fetching if there are no results? Also, if the task is only started, I imagine we would want to get test results again in the future if the task has finished since they might have changed?

Reply:
I think there is a case where we want to merge legacy test results and test results from the test results collection. We don't set populated until the task is finished (see L2068).

Reply:
Are you referring to `if !evergreen.IsFinishedTaskStatus(t.Status) && t.Status != evergreen.TaskStarted {` ? So we do set it if the task is started but not finished.